src/packet_analysis/protocol/llc/LLC.cc

```diff
@@ -0,0 +1,32 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#include "zeek/packet_analysis/protocol/llc/LLC.h"
+
+using namespace zeek::packet_analysis::LLC;
+
+LLCAnalyzer::LLCAnalyzer() : zeek::packet_analysis::Analyzer("LLC") { }
+
+bool LLCAnalyzer::AnalyzePacket(size_t len, const uint8_t* data, Packet* packet)
+	{
+	// An LLC header is at least 3 bytes, check for that first.
```

Comment:
With the suggested changes to the Ethernet analyzer, I think the overall packet length could be validated as well.

Reply:
I think the length check in the ethernet analyzer is wrong as it is, to be honest:

```
		if ( 16 >= len )
			{
			Weird("truncated_ethernet_frame", packet);
			return false;
			}
```

We already made that comparison earlier. This should be checking whether the remaining data fits, which is the value of `protocol` that was pulled out earlier.